app/modules/entities/views/entities_list_adder.js

```diff
@@ -74,6 +74,12 @@ export default Marionette.CompositeView.extend({
       claims['wdt:P123'] = parentModel.get('claims.wdt:P123')
     }
 
+    if (parentModel.get('claims.wdt:P31')?.length === 1) {
+      const parentP31 = parentModel.get('claims.wdt:P31.0')
```

Comment:
I would be for reducing to a maximum extent the use of those oneliners `?`, as, to me, they hide complexity by easing access to deep objects

```suggestion
    const P31values = parentModel.get('claims.wdt:P31')
    if (P31values && P31values.length === 1) {
      const parentP31 = P31values[0]
```

Reply:
I guess it depends of the cases: here, your suggestion does seem clearer (correction made in 8beb8c0c), but we have been using `?` in coffeescript for years and were quite happy about it (it was even listed as one of the [reasons to stay with coffee](https://github.com/inventaire/inventaire/issues/72)); so I wouldn't agree to make it a general rule

Reply:
Its not only complexity, its also prone to inconsistencies and silent errors. Can we say at least agree that we cannot use this operator if the result is stored in a variable (so a `if` like this one is fine) ?

Reply:
a good case provided by the server decaffeination:
```coffee
publicationYear = entity.claims['wdt:P577']?[0]?.split('-')[0]
```
was decaffeinated as
```js
const publicationDateClaim = claims['wdt:P577'] && claims['wdt:P577'][0]
const publicationYear = publicationDateClaim && publicationDateClaim.split('-')[0]
```
but once we update the server engine to NodeJS >= 14, we could rewrite it like this
```js
const publicationYear = entity.claims['wdt:P577']?.[0]?.split('-')[0]
```
I do find it more readable

Reply:
Sure its more readable, no doubt on that. Im saying that this operator *can* make results inconsistent and throw silent errors, which is much more problematic that its readability imo
https://dev.to/macsikora/what-is-wrong-with-optional-chaining-and-how-to-fix-it-3nno